contrib/NeonVM/neonvm.c

```diff
@@ -746,16 +898,26 @@ void exec_CALLP(struct tagTExecutor *self)
         strcat(s, "]");
         pop(self->stack);
         push(self->stack, cell_fromString(s));
+        free(s);
     } else if (strcmp(func, "concat") == 0) {
-        char *b = _strdup(top(self->stack)->string); pop(self->stack);
-        char *a = _strdup(top(self->stack)->string); pop(self->stack);
+        char *b = strdup(top(self->stack)->string); pop(self->stack);
+        char *a = strdup(top(self->stack)->string); pop(self->stack);
         char *n = malloc(strlen(a) + strlen(b) + 1);
         strcpy(n, a);
         strcat(n, b);
         push(self->stack, cell_fromString(n));
         free(a);
         free(b);
         free(n);
+    } else if (strcmp(func, "number__toString") == 0) {
+        BID_UINT128 v = top(self->stack)->number; pop(self->stack);
+        push(self->stack, cell_fromString(number_to_string(v)));
+    } else if (strcmp(func, "string__append") == 0) {
+        char *b = strdup(top(self->stack)->string); pop(self->stack);
```

Comment:
After doing this strdup, who frees b? I suggest you don't need to strdup at all, because you're only reading from the string.